### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by xannz**
This PR is almost a complete rewrite of the ssl-termination charm. Fixing multiple issues:
#12 
#13 
#15 
#16 

The charm also uses the [nginx-config](https://github.com/tengu-team/layer-nginx-config) layer.

### CODE REVIEW

File: reactive/ssl_termination_proxy.py [hunk 1]
```diff
@@ -58,15 +88,25 @@ def get_certificate_requests():
         old_requests = unitdata.kv().get('sslterm.cert-requests', [])
         delete_old_certs(old_requests, cert_requests)
         unitdata.kv().set('sslterm.cert-requests', cert_requests)
-        lets_encrypt.set_requested_certificates(cert_requests)
+        pre_cert_requests = prepare_cert_requests(cert_requests)
+        lets_encrypt.set_requested_certificates(pre_cert_requests)
+        NginxConfig().delete_all_config(NginxModule.HTTP, 'ssl-termination')
         set_flag('ssl-termination.waiting')
+<<<<<<< HEAD
```
Comment (kwmonroe) [inline reactive/ssl_termination_proxy.py:95]: Sorry to comment on a merged PR, but it looks like you have a dirty merge here.

File: templates/encrypt.nginx.jinja2 [hunk 2]
```diff
@@ -55,14 +55,20 @@ server {
...
+{% if nginx_config %}
+    {%- for cfg in nginx_config %}
+        {{cfg}};
+    {%- endfor -%}
+{% endif %}
+
         {% if auth_basic %}
         # Don't forward auth to backend service
-        proxy_set_header Authorization "";
+        proxy_set_header Authorization "";
```
Comment (kwmonroe) [inline templates/encrypt.nginx.jinja2:66]: trailing whitespace

File: templates/encrypt.nginx.jinja2 [hunk 3]
```diff
@@ -55,14 +55,20 @@ server {
...
         # Don't forward auth to backend service
-        proxy_set_header Authorization "";
+        proxy_set_header Authorization "";        
 
         limit_except OPTIONS {
             auth_basic "Restricted Content";
             auth_basic_user_file {{htpasswd}};
         }
-        {% endif %}
+        {% endif %}
```
Comment (kwmonroe) [inline templates/encrypt.nginx.jinja2:72]: trailing whitespace

**Review by kwmonroe — COMMENTED** [review]
I spotted a few minor issues with the recent merge.

**Comment by xannz** [comment]
@kwmonroe  Thanks for catching those, I'll get those fixed!

**Pull request merged**